### PR TITLE
Fix user

### PR DESCRIPTION
**Pull request by maria-j-k**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "python.linting.flake8Enabled": true,
+    "python.linting.enabled": true,
+    "python.linting.pylintEnabled": false,
+    "python.linting.pylamaEnabled": false
+}
```
Comment (nitz14) [inline .vscode/settings.json]: vs code configuration shouldn't be add to github repo

File: trello/settings.py [hunk 2]
```diff
@@ -9,11 +9,12 @@
 For the full list of settings and their values, see
 https://docs.djangoproject.com/en/3.2/ref/settings/
 """
-
-from pathlib import Path
+from datetime import timedelta
 import environ
```
Comment (nitz14) [inline trello/settings.py:12]: pep8 -> sort import

File: users/views.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    })
+
+
+class UserViewSet(viewsets.ModelViewSet):
+    queryset = User.objects.all()
+    serializer_class = UserSerializer
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        if serializer.is_valid():
```
Comment (nitz14) [inline users/views.py:29]: We don't need to do if else in serializer 

https://www.django-rest-framework.org/api-guide/serializers/#validation

File: users/views.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                            headers=headers)
+        else:
+            logger.error('Somebody has entered an invalid email!')
+            return Response('You need to provide a valid email address',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    @action(detail=False, methods=['GET'])
+    def activate(self, request):
+        user_id = request.query_params.get('user_id', '')
+        serializer = self.get_serializer(data={'id': 'user_id'})
```
Comment (nitz14) [inline users/views.py]: This like is correct in your opinion? Did you test it?

File: users/views.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        serializer.is_valid(raise_exception=True)
+        confirmation_token = request.query_params.get('token', '')
+        user = get_object_or_404(self.get_queryset(), pk=user_id)
+        if not default_token_generator.check_token(user, confirmation_token):
+            return Response('''Token is invalid or expired.
+            Please request another confirmation email by signing in.''',
+                            status=status.HTTP_400_BAD_REQUEST)
+        user.is_active = True
+        user.save()
```
Comment (nitz14) [inline users/views.py]: we speak about saving whole object.

File: users/views.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            logger.error('Somebody has entered an invalid email!')
+            return Response('You need to provide a valid email address',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    @action(detail=False, methods=['GET'])
+    def activate(self, request):
+        user_id = request.query_params.get('user_id', '')
+        serializer = self.get_serializer(data={'id': 'user_id'})
+        serializer.is_valid(raise_exception=True)
+        confirmation_token = request.query_params.get('token', '')
```
Comment (nitz14) [inline users/views.py]: maybe validation of that token can be done in serializer?

**Review by nitz14 — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 1 above)
Comment (maria-j-k) [reply .vscode/settings.json]: Thanks, I forgot to add it to .gitignore before opening folder in vs code

File: trello/settings.py (hunk 2 above)
Comment (maria-j-k) [reply trello/settings.py:12]: But it's sorted... I've sorted it in first commit in this branch...

File: users/views.py (hunk 3 above)
Comment (maria-j-k) [reply users/views.py:29]: I now about the syntax serializer.is_valid(raise_exception=True). I didn't know how to make logg with this syntax, that's why I've chosen if - else block.

File: users/views.py (hunk 4 above)
Comment (maria-j-k) [reply users/views.py]: corrected

File: users/views.py (hunk 5 above)
Comment (maria-j-k) [reply users/views.py]: done

File: users/views.py (hunk 6 above)
Comment (maria-j-k) [reply users/views.py]: done

File: trello/settings.py (hunk 2 above)
Comment (nitz14) [reply trello/settings.py:12]: 1. absolute imports than from

Comment (maria-j-k) [reply trello/settings.py:12]: done

**Review by nitz14 — APPROVED** [review]
LF

**Pull request merged**